src/control/lib/control/pool.go

```diff
@@ -717,3 +700,141 @@ func PoolReintegrate(ctx context.Context, rpcClient UnaryInvoker, req *PoolReint
+		UUID string `json:"uuid"`
+		// Label is an optional human-friendly identifier for a pool.
+		Label string `json:"label,omitempty"`
+		// ServiceReplicas is the list of ranks on which this pool's
+		// service replicas are running.
+		ServiceReplicas []system.Rank `json:"svc_reps"`
+
+		// TargetsTotal is the total number of targets in pool.
+		TargetsTotal uint32 `json:"targets_total"`
+		// TargetsDisabled is the number of inactive targets in pool.
+		TargetsDisabled uint32 `json:"targets_disabled"`
+
+		// QueryError reports an RPC error returned from a query.
+		QueryErrorMsg string `json:"query_error_msg"`
+		// QueryStatus reports any DAOS error returned from a query
+		// operation.
+		QueryStatus int32 `json:"query_status"`
+
+		// Usage contains pool usage statistics for each storage tier.
+		Usage []*PoolTierUsage `json:"usage"`
+	}
+)
+
+func (p *Pool) setUsage(pqr *PoolQueryResp) {
+	nvmeTotal := pqr.Nvme.Total
+	scmTotal := pqr.Scm.Total
+
+	scmSpread := pqr.Scm.Max - pqr.Scm.Min
+	scmImbalance := float64(scmSpread) / (float64(scmTotal) / float64(pqr.ActiveTargets))
+	nvmeSpread := pqr.Nvme.Max - pqr.Nvme.Min
+	nvmeImbalance := float64(nvmeSpread) / (float64(nvmeTotal) / float64(pqr.ActiveTargets))
+
+	p.Usage = append(p.Usage,
+		// scm is tier 0
+		&PoolTierUsage{
+			TierName:  "SCM",
+			Size:      scmTotal,
+			Free:      pqr.Scm.Free,
+			Imbalance: uint32(scmImbalance * 100),
+		},
+		// nvme is tier 1
+		&PoolTierUsage{
+			TierName:  "NVME",
+			Size:      nvmeTotal,
+			Free:      pqr.Nvme.Free,
+			Imbalance: uint32(nvmeImbalance * 100),
+		})
+}
+
+// ListPoolsReq contains the inputs for the list pools command.
+type ListPoolsReq struct {
+	unaryRequest
+	msRequest
+}
+
+// ListPoolsResp contains the status of the request and, if successful, the list
+// of pools in the system.
+type ListPoolsResp struct {
+	Status int32   `json:"status"`
+	Pools  []*Pool `json:"pools"`
+}
+
+// ListPools fetches the list of all pools and their service replicas from the
+// system.
+func ListPools(ctx context.Context, rpcClient UnaryInvoker, req *ListPoolsReq) (*ListPoolsResp, error) {
+	req.setRPC(func(ctx context.Context, conn *grpc.ClientConn) (proto.Message, error) {
+		return mgmtpb.NewMgmtSvcClient(conn).ListPools(ctx, &mgmtpb.ListPoolsReq{
+			Sys: req.getSystem(rpcClient),
+		})
+	})
+	rpcClient.Debugf("DAOS system list-pools request: %+v", req)
+
+	ur, err := rpcClient.InvokeUnaryRPC(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := new(ListPoolsResp)
+	if err := convertMSResponse(ur, resp); err != nil {
+		return nil, err
+	}
+
+	// issue query request and populate usage statistics for each pool
+	for _, p := range resp.Pools {
+		rpcClient.Debugf("Fetching details for discovered pool: %v", p)
+
+		resp, err := PoolQuery(ctx, rpcClient, &PoolQueryReq{UUID: p.UUID})
+		if err != nil {
+			p.QueryErrorMsg = err.Error()
+			if p.QueryErrorMsg == "" {
+				p.QueryErrorMsg = "unknown error"
+			}
```

Comment:
How is that factored into the dmg list output? We should make sure to still print the pool list and have a ERR in the column that we cannot display.

Reply:
in discussion with @mjmac and @johannlombardi decided not to print errored pool entry in list but print the rest and print any failed pool query before the table i.e.
```
Query on pool "00000002" unsuccessful, error: "stats unavailable"
Query on pool "three" unsuccessful, status: "DER_UNINIT"

Pool Size   Used Imbalance Disabled
---- ----   ---- --------- --------
one  6.0 TB 83%  12%       0/16

```